python/common/org/python/stdlib/datetime/DateTime.java

```diff
@@ -0,0 +1,539 @@
+                return org.python.types.Bool.getBool(false);
+            }
+
+            if (((org.python.types.Bool)this.__microsecond__().__lt__(((DateTime) other).__microsecond__())).value){
+                return org.python.types.Bool.getBool(true);
+            } else if (((org.python.types.Bool)this.__microsecond__().__gt__(((DateTime) other).__microsecond__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            return org.python.types.Bool.getBool(false);
+        }
+        return org.python.types.NotImplementedType.NOT_IMPLEMENTED;
+    }
+
+    @org.python.Method(
+        __doc__ = "Return self <= dateTime.",
+        args = {"other"}
+    )
+    public org.python.Object __le__(org.python.Object other){
+        if (other instanceof DateTime){
+            if (((org.python.types.Bool)this.__year__().__gt__(((DateTime) other).__year__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__month__().__gt__(((DateTime) other).__month__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__day__().__gt__(((DateTime) other).__day__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__hour__().__gt__(((DateTime) other).__hour__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__minute__().__gt__(((DateTime) other).__minute__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__second__().__gt__(((DateTime) other).__second__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__microsecond__().__gt__(((DateTime) other).__microsecond__())).value){
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__year__().__le__(((DateTime) other).__year__())).value &&
+                ((org.python.types.Bool)this.__month__().__le__(((DateTime) other).__month__())).value &&
+                ((org.python.types.Bool)this.__day__().__le__(((DateTime) other).__day__())).value &&
+                ((org.python.types.Bool)this.__hour__().__le__(((DateTime) other).__hour__())).value &&
+                ((org.python.types.Bool)this.__minute__().__le__(((DateTime) other).__minute__())).value &&
+                ((org.python.types.Bool)this.__second__().__le__(((DateTime) other).__second__())).value &&
+                ((org.python.types.Bool)this.__microsecond__().__le__(((DateTime) other).__microsecond__())).value){
+                return org.python.types.Bool.getBool(true);
```

Comment:
This code block is not necessary, the method works without it. We think removing it improves the readability of the code.